src/js/components/forms/Dropdown.jsx

```diff
@@ -56,6 +56,18 @@ export class Dropdown extends React.Component {
         return this.state.selectedOption;
     }
 
+    selectOption(option) {
```

Comment:
Yeah, good topic for discussion here. I would like to use props wherever possible, although the "interactive components" have a little wrinkle. I want to be careful to avoid forcing users of the components to have boilerplate code if it's possible. Using `Dropdown` as an example..

Dropdown has a `defaultOption` prop and a `selectedOption` state var. We need need to support three use cases:
1. Allow the Dropdown's "selected option" to be set during initial render. This is currently done with `defaultOption` prop.
2. Make sure the Dropdown's selected option is updated when the user picks an item in the list. This is done by updating the `selectedOption` state var when the user changes the item.
3. Allow the selected option to be to set programmatically at any time. This was going to be done via that `selectOption` method.

If the developer doesn't need to worry about 1 or 3, then it seems nice that the Dropdown tracks the item using its own internal state, so there isn't a need for tracking the selected option in the parent component's state (or elsewhere).

But maybe it makes more sense just to create a `selectedOption` prop on Dropdown. There seems to be a widely-accepted rule in the React world that component never update their own props. So for use case 2, it seems it the developer would need to track selectedOption in the state of the parent component, and update it via the change handler... which seems tedious, but maybe that makes the most sense.

Sorry for the length... does that make sense? Let me know if you have an inclination either way.